src/hooks/screen/useChatScreenHook.ts

```diff
@@ -64,13 +67,6 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
     }
   );
 
-  React.useEffect(() => {
-    if (getAllMessages.data) {
-      setChats(getAllMessages.data);
-      setIsLoadingFetchAllMessage(false);
-    }
-  }, [getAllMessages.data]);
-
   const initChatData = async () => {
     if (!localDb || !selectedChannel) return;
     if (!otherListener[`chat_${selectedChannel?.id}`]) return;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChatData` function checks if `localDb`, `selectedChannel`, and `otherListener[`chat_${selectedChannel?.id}`]` are not null or undefined. However, it doesn't handle the case where these values might be null or undefined. It would be better to throw an error or return a default value when these conditions are not met.
```diff
   const initChatData = async () => {
     if (!localDb || !selectedChannel) throw new Error("Local DB or selected channel is not defined");
     if (!otherListener[`chat_${selectedChannel?.id}`]) throw new Error(`Other listener for chat_${selectedChannel?.id} is not defined`);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->